exec/plugin-commands-script-runners/src/exec.ts

```diff
@@ -196,15 +200,23 @@ export async function handler (
   const workspacePnpPath = opts.workspaceDir && existsPnp(opts.workspaceDir)
 
   let exitCode = 0
-  const prependPaths = [
-    './node_modules/.bin',
-    ...opts.extraBinPaths,
-  ]
+  const mapPrefixToPrependPaths: Record<ProjectRootDir, string[]> = {}
+  await Promise.all(chunks.flat().map(async prefix => {
+    const executionEnv = await prepareExecutionEnv(opts, {
+      extraBinPaths: opts.extraBinPaths,
+      executionEnv: opts.selectedProjectsGraph[prefix]?.package.manifest.pnpm?.executionEnv,
+    })
+    mapPrefixToPrependPaths[prefix] = [
+      './node_modules/.bin',
```

Comment:
This is probably better?

```suggestion
      path.join(prefix, 'node_modules', '.bin'),
```

Reply:
It wasn't better (fails the "path delimiter" test).